### PR TITLE
CC-33 Create new Menu component

### PR DESCRIPTION
**Pull request by 17Sergey**
[CC-33](https://pavelpiasetski.atlassian.net/browse/CC-33)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ function App() {
   return (
     <div className="App">
       <Welcome welcomeText={`Храм в честь иконы Божьей Матери "Неупиваемая Чаша".`} />
+      <Welcome welcomeText={`Храм в честь иконы Божьей Матери "Неупиваемая Чаша".`} />
```
Comment (17Sergey) [inline src/App.tsx:8]: Why do you do that?

Comment (17Sergey) [reply src/App.tsx:8]: Because I can

**Comment by PavelPyasecky** [comment]
> ### ❌ Deploy Preview for _musical-meringue-ae05e3_ failed.
> Name 	Link
> 🔨 Latest commit 	
> 🔍 Latest deploy log 	https://app.netlify.com/sites/musical-meringue-ae05e3/deploys/65a41b253ee3e19908fcbde9

Probably we have to repair this. 
image

File: package.json [hunk 2]
```diff
@@ -28,12 +29,16 @@
...
+    "autoprefixer": "^10.4.16",
     "eslint": "^8.0.0",
     "eslint-config-react-app": "^7.0.1",
     "eslint-plugin-prettier": "^4.2.1",
     "jsdom": "^21.1.0",
+    "postcss": "^8.4.33",
     "prettier": "^2.7.1",
     "prettier-config-nick": "^1.0.2",
+    "styled-components": "^6.1.6",
+    "tailwindcss": "^3.4.0",
```
Comment (PavelPyasecky) [inline package.json]: You can leave the link in a ticket to a new libraries which you gonna use. (like https://tailwindcss.com/)

**Comment by PavelPyasecky** [comment]
It's a good habit using valuable and a bit more meaningful name for the commit name Other developers would like to understand the changes without looking the the code changes that were made.
image

https://www.aleksandrhovhannisyan.com/blog/atomic-git-commits/

File: src/components/Sidebar/MenuItems/MenuItems.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            menuItem => {
+                return <MenuItem
+                            key={menuItem.id}
+                            id={menuItem.id}
+                            name={menuItem.name}
+                            icon={menuItem.icon}
+                            submenu={menuItem.submenu}
+                            submenuExpanded={menuItem.submenuExpanded}
+                            expanded={expanded}
+                            isActive={menuItem.id === activeItemId ? true : false}
```
Comment (PavelPyasecky) [inline src/components/Sidebar/MenuItems/MenuItems.tsx]: Let's just simplify the condition to `menuItem.id === activeItemId`

File: src/components/Sidebar/MenuItems/MenuItems.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            }
+        )
+    }
+
+    return (
+        <ul className="flex-1 mt-5 mb-3">
+            {createMenuItems()}
+        </ul>
+    );
+}
```
Comment (PavelPyasecky) [inline src/components/Sidebar/MenuItems/MenuItems.tsx:32]: A good practice is to add an empty line to each file. https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file

File: src/components/Sidebar/MenuItems/MenuItems.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+export const MenuItems: React.FC<MenuItemsProps> = ({ expanded }) => {
+
+    const menuItems = useAppSelector(selectMenuItems);
+    const activeItemId = useAppSelector(selectActiveItemId);
+
+    const createMenuItems = (): Array<ReactNode> => {
+        return menuItems.map(
+            menuItem => {
+                return <MenuItem
```
Comment (PavelPyasecky) [inline src/components/Sidebar/MenuItems/MenuItems.tsx]: I suppose we can reduce the amount of `return` instructions to reduce mental load of reviewers. Every time I see the return instruction I think about the end of the component.

File: src/components/Sidebar/MenuItems/MenuItems.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                            submenuExpanded={menuItem.submenuExpanded}
+                            expanded={expanded}
+                            isActive={menuItem.id === activeItemId ? true : false}
+                        />
+            }
+        )
+    }
+
+    return (
+        <ul className="flex-1 mt-5 mb-3">
```
Comment (PavelPyasecky) [inline src/components/Sidebar/MenuItems/MenuItems.tsx]: It looks like inconsistent approach to the styling methodology. But will return to that in the future :)

File: tailwind.config.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  content: [
+    "./index.html",
+    "./src/**/*.{js,ts,jsx,tsx}",
+  ],
+  theme: {
+    extend: {
+      colors: {
+        "sidebar-blue": "#0075BA",
+        // "sidebar-blue-transparrent": "rgb(230,247,255)",
+        // "sidebar-blue-transparrent": "rgba(0, 117, 186, 0.3)",
```
Comment (PavelPyasecky) [inline tailwind.config.js]: Why do we need to commit commented code?

**Comment by PavelPyasecky** [comment]
@17Sergey 
The work which has been done is REALLY awesome! :100: 
Keep up your work and we will reach our aim even faster than think :1st_place_medal: 

image

**Comment by 17Sergey** [comment]
Hello, fixed a bug that prevented from build

**Review by PavelPyasecky — APPROVED** [review]

**Review by PavelPyasecky — APPROVED** [review]

**Pull request merged**